src/main/java/net/snowflake/client/core/bind/BindUploader.java

```diff
@@ -161,19 +151,26 @@ private synchronized String synchronizedTimestampFormat(String o) {
    * @param stageDir the stage path to upload to
    * @return BindUploader instance
    */
-  public static synchronized BindUploader newInstance(SFSession session, String stageDir) {
+  public static synchronized BindUploader newInstance(SFBaseSession session, String stageDir) {
     return new BindUploader(session, stageDir);
   }
 
+  /** Wrapper around upload() with default compression to true. */
+  public void upload(Map<String, ParameterBindingDTO> bindValues)
+      throws BindException, SQLException {
+    upload(bindValues, true);
+  }
+
   /**
    * Upload bind parameters via streaming. This replaces previous function upload function where
    * binds were written to a file which was then uploaded with a PUT statement.
    *
    * @param bindValues the bind map to upload
+   * @param compressData whether or not to compress data
    * @throws BindException
    * @throws SQLException
    */
-  public void upload(Map<String, ParameterBindingDTO> bindValues)
+  public void upload(Map<String, ParameterBindingDTO> bindValues, boolean compressData)
```

Comment:
Note: modified because current stored-proc file-transfer agent implementation does not support compressing data streams, so `StoredProcSFStatement` has a way to disable it when performing array binds.

Reply:
Do user directly interact with this function? Do we need to worry about this being breaking change?

Reply:
I added the single-argument function with the original signature up top as a wrapper of this function to avoid this.